usaspending_api/agencies/management/commands/load_agencies.py

```diff
@@ -0,0 +1,276 @@
+                messages.append("Row number {:,} has a SUBTIER CODE but no SUBTIER NAME".format(agency.row_number))
+            if agency.is_frec is True and agency.frec is None:
+                messages.append("Row number {:,} is marked as IS_FREC but has no FREC".format(agency.row_number))
+            if agency.is_frec is not True and agency.cgac_agency_code is None:
+                messages.append(
+                    "Row number {:,} is not marked as IS_FREC but has no CGAC AGENCY CODE".format(agency.row_number)
+                )
+            if agency.cgac_agency_code and len(agency.cgac_agency_code) != 3:
+                messages.append(
+                    "Row number {:,} has CGAC AGENCY CODE that is not 3 characters long ({})".format(
+                        agency.row_number, agency.cgac_agency_code
+                    )
+                )
+            if agency.frec and len(agency.frec) != 4:
+                messages.append(
+                    "Row number {:,} has FREC that is not 4 characters long ({})".format(agency.row_number, agency.frec)
+                )
+            if agency.subtier_code and len(agency.subtier_code) != 4:
+                messages.append(
+                    "Row number {:,} has SUBTIER CODE that is not 4 characters long ({})".format(
+                        agency.row_number, agency.subtier_code
+                    )
+                )
+
+        if messages:
+            for message in messages:
+                logger.error(message)
+            raise RuntimeError(
+                "{:,} problem(s) have been found with the agency file.  See log for details.".format(len(messages))
+            )
+
+    def _import_raw_agencies(self):
+        with self.connection.cursor() as cursor:
+            execute_values(
+                cursor.cursor,
+                """
+                    insert into temp_load_agencies_raw_agency (
+                        row_number,
+                        cgac_agency_code,
+                        agency_name,
+                        agency_abbreviation,
+                        frec,
+                        frec_entity_description,
+                        frec_abbreviation,
+                        subtier_code,
+                        subtier_name,
+                        subtier_abbreviation,
+                        toptier_flag,
+                        is_frec,
+                        user_selectable,
+                        mission,
+                        website,
+                        congressional_justification,
+                        icon_filename,
+                        include_toptier_without_subtier
+                    ) values %s
+                """,
+                self.agencies,
+                page_size=len(self.agencies),
+            )
+            if cursor.rowcount > -1:
+                logger.info("{:,} rows affected".format(cursor.rowcount))
+
+    def _run_steps(self):
```

Comment:
This whole function looks like a great candidate for reuse; I would look into moving this and the ProccesingStep class into a common area for anything we are using the sql approach for.

Reply:
While I would love to do this, I'm going to have to pass for this ticket.  I tinkered a little and I think it's bigger than a breadbox.